client/lxd.go

```diff
@@ -492,3 +493,196 @@ func (r *ProtocolLXD) getUnderlyingHTTPTransport() (*http.Transport, error) {
 		return nil, fmt.Errorf("Unexpected http.Transport type, %T", r)
 	}
 }
+
+// tryForwardTargetOpToRemoteOp generalizes the forwarding of a target operation for API request calls.
```

Comment:
Please can you break stuff like this, that isn't strictly related to the task at hand (instance rebuild) into separate commits that are earlier on in the PR, and then additional commits to add anything specific to instance rebuild. It makes it easier to reason about each change and whether its related to the task in hand.

My feeling with this function is that its too general, and I'm not sure what its trying to solve, but will wait to see it in isolation to decide.

Reply:
Yeah that's what I thought too. Plus, when I look at your other comments, I realise that using this function will be a bit of a mess. Let's not use this function. It'll become too complex with interfaces everywhere...